### ISSUE TITLE
[FE] My VA: Clean up feature flag code on mock server

### ISSUE THREAD

**Issue by ajakabcin** [department-of-veterans-affairs/va.gov-team#65042]
## Background
Currently the way our [feature toggles are set up on our mock server](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/personalization/common/mocks/feature-toggles/index.js) is clunky and doesn't feel sustainable the more features we add. 

If we model our mock feature toggle file after [profile's](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/personalization/profile/mocks/endpoints/feature-toggles/index.js), we can just update or add feature toggles in one place, rather than two. This will improve developer quality-of-life in the long run. 

## Tasks
- [x] meet with Platform engineering CoP 
- [x] update `src/applications/personalization/common/mocks/feature-toggles/index.js`
- [x] update src/applications/personalization/common/mocks/server.js

## Acceptance Criteria
- [x] Feature toggle file is updated
- [x] Feature toggles still behave correctly

## Validation
- _Who can validate this ticket? (FE, BE, Design, PM)?_ Devs: @dellerbie and @adamwhitlock1 
- _How can this work be validated?_ confirm that all feature toggles are working as intended on mock server

## Resources
- [Adam's note + suggestions](https://github.com/department-of-veterans-affairs/vets-website/pull/24735#discussion_r1268315096)
- [feature toggle mocks on /profile/](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/personalization/profile/mocks/endpoints/feature-toggles/index.js)

**Comment by allisonlu** [comment]
@ajakabcin just added more info to the description. I'll remove the **needs-refinement** label as well!

**Comment by DanielMiller-agile6** [comment]
test

**Comment by ajakabcin** [comment]
@allisonlu when you pick this up can you have @DanielMiller-agile6 shadow/follow along? Thank you!

**Comment by ajakabcin** [comment]
@DanielMiller-agile6 if you need assistance on this give @allisonlu a shout!

**Comment by DanielMiller-agile6** [comment]
PR: https://github.com/department-of-veterans-affairs/vets-website/pull/27459

**Comment by DanielMiller-agile6** [comment]
I lost internet and power with the ice storm here, over the past few days, and had kids cooped up at home with no school, so this ticket is carying over.  That said, i think we should close out this ticket and create a new one, since work on this revealed a few additional problems we'll need to consider if we want to refactor our feature flag mocks to simplify or improve developer & product-owner experience.

Adam pointed out an issue with our local mock server that I had missed.  I had only run cypress tests, which seem to work fine with my changes, but the local mocking server runs in common js module mode, which means we can't synchronously import platform's master list of flags and client<->server name mappings.  This would break the local development experience for lots of devs, so we'd need to work around this if we want to move forward with the suggested implementation from the pull request.

We've got a few options to consider for a new ticket, so maybe we can discuss this in refinement, retro or planning. The options I can imagine so far are:
- synchronously read platform master list as text and regexp parse a json string. not ideal, but a workable stop-gap solution if we want to merge our existing PR as-mostly-is.
- discover if mocker-js has any way to support async es module loading for tthe mock config file.  i couldn't get this to work after a few quick tests, but perhaps there's a way. this would be safer than reading and parsing the module as text like previous idea, but may be impossible due to mocker-api implementation.
- see if platform team could manage this master list as a .json file, instead of .js.  If they did this, they could also provide 3 additional modules, a .cjs version (common js module exports), a .mjs version (es module exports), and a .js version (default environment and/or bundler settings). each module would just import the JSON string, and export, sam as before, but devs and product owners would only ever need to manually touch the json file. i think this might be ideal, but platform changes might take a long time so we'd need wait on that before continuing on this ticket.
- we could also try to use the snake_case method like profile team, but we discovered that we'd needed to rename some active feature flags in order to do this.  not all of our feature flag names are straight up camelCase->snake_case mappings. to do this, we'd need to change all affected flag names across codebase which I'd like to avoid, but also maybe a stop-gap option.  ideally, we shouldn't have to rely on a naming convention for this mapping at all for this, since it still requires us to define the mapping in multiple places, instead of a single location with the platform's master list.

**Comment by DanielMiller-agile6** [comment]
moving this ticket to backlog for now until we can chat and determine direction or propose platform changes.

**Comment by ajakabcin** [comment]
Moving to next sprint as we still need to chat with Clint and platform folks

**Comment by DanielMiller-agile6** [comment]
Discussed with and got approval from platform team, so we're good to go with our changes.  Just merged the PR, so I'll try to verify tomorrow when things get to staging environment.

**Comment by DanielMiller-agile6** [comment]
verified deploy to staging, and login myva worked as expected.  since there aren't any user-facing changes to this (just effects local and cypress mocking environments). I also pulled down latest from main, and ran mocker-api locally, using new code to verify.
